airbyte/_util/temp_files.py

```diff
@@ -21,12 +23,15 @@ def as_temp_files(files_contents: list[dict | str]) -> Generator[list[str], Any,
     """Write the given contents to temporary files and yield the file paths as strings."""
     temp_files: list[Any] = []
     try:
+        temp_dir = OVERRIDE_TEMP_DIR
+
```

Comment:
@coderabbitai Thank you for the feedback. Please observe the other change with the `constants.py` - I am defaulting the value to `None`, in case the environment variable was not supplied.

And the reason for doing that is simple, since that way, we wouldn't want to break the existing flow - i.e. the `dir`'s argument to not be supplied/ `None`!